src/modules/paging/oneSafe.js

```diff
@@ -0,0 +1,30 @@
+/* eslint-disable import/prefer-default-export */
+// @ts-check
+import { okAsync, errAsync, ResultAsync } from 'neverthrow';
+import { NoResults, OpenAPIError, TooManyResults } from '../../error';
+
+/**
+ * @param {any} req
+ * @param {{
+ *  searchRequest: (req: any, args: any, options: any) => Promise<{items: any[]}>,
+ *  args: Record<string, unknown>,
+ *  options?: Record<string, unknown>
+ * }} args
+ * @returns {ResultAsync<any, NoResults | TooManyResults>}
+ */
+export function oneSafe(req, { searchRequest, args = {}, options = {} }) {
```

Comment:
what about some tests? good to document how to use the function?

Reply:
good point, i'll add some